libraries/stdlib/src/kotlin/annotations/NativeAnnotations.kt

```diff
@@ -16,3 +16,30 @@ package kotlin.native
 @Retention(AnnotationRetention.BINARY)
 @OptionalExpectation
 public expect annotation class CName(val externName: String = "", val shortName: String = "")
+
+/**
+ * This annotation marks the experimental [ObjCName] annotation.
+ */
+@RequiresOptIn
+@Target(AnnotationTarget.ANNOTATION_CLASS)
+@Retention(AnnotationRetention.BINARY)
+@MustBeDocumented
+@OptionalExpectation
+public expect annotation class ExperimentalObjCName()
+
+/**
+ * Instructs the Kotlin compiler to use a custom Objective-C and/or Swift name for this class, property, parameter or function.
+ * @param exact specifies if the name of a class, object or interface should be interpreted as the exact name.
+ * E.g. the compiler won't add a top level prefix or the outer class names to exact names.
+ */
+@Target(
+    AnnotationTarget.CLASS,
+    AnnotationTarget.PROPERTY,
+    AnnotationTarget.VALUE_PARAMETER,
+    AnnotationTarget.FUNCTION
+)
+@Retention(AnnotationRetention.BINARY)
+@MustBeDocumented
```

Comment:
Well the annotation manipulates the public API (in this case the ObjC/Swift API instead of the Kotlin API).

While it's true that from ObjC/Swift you would only see the modified names, that isn't the case if you view the Kotlin documentation (e.g. generated by Dokka).

If you look at this from either ObjC/Swift or Kotlin the annotation isn't that important and we could probably drop the `@MustBeDocumented`. However IMO it is useful information when you try to understand the interop.

Reply:
Ok, let's just add it to an ObjCExport `@MustBeDocumented` exclude list then.

Reply:
Recently, we've merged the support for `@MustBeDocumented`.
So please add `@ObjCName` annotation to the exclude list: https://github.com/JetBrains/kotlin/blob/eaad75fd52ac4a0b0deca02d1aaa9b4872c4dcfb/kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportHeaderGenerator.kt#L811